### PR TITLE
MWPW-146209 [MILO][MEP] MEP param override does not work if manifest sent by Target and pzn and chosen column not in Target

### PR DESCRIPTION
**Pull request by vgoodric**
Steps to reproduce:
1. Create a manifest
3. Create a page that references that manifest in its personalization metadata
4. Create a Target activity that serves the same manifest to the page
5. Update the manifest to have a new column but don't update the Target offer
6. Try to use the MEP button to spoof the new column

Solution: don't check if the selected variant should be overridden until the manifests have been consolidated in cleanAndSortManifestList function

Example page setup that is sending an old version of manifest via Target and is trying to spoof a new column not sent by Target:
 - Before: https://main--cc--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q2/mepparamtarget/?at_preview_token=vq2HGezS8xc9gv4--gb5Ug&at_preview_index=1_1&at_preview_listed_activities_only=true&mep=%2Fdrafts%2Fvgoodrich%2Ffragments%2F2024%2Fq2%2Fmepparamtarget%2Fmanifest.json--photoshop-any
 -  After: https://main--cc--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q2/mepparamtarget/?milolibs=mepparamtarget&at_preview_token=vq2HGezS8xc9gv4--gb5Ug&at_preview_index=1_1&at_preview_listed_activities_only=true&mep=%2Fdrafts%2Fvgoodrich%2Ffragments%2F2024%2Fq2%2Fmepparamtarget%2Fmanifest.json--photoshop-any

Resolves: [MWPW-145303](https://jira.corp.adobe.com/browse/MWPW-145303)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://mepparamtarget--milo--adobecom.hlx.page/?martech=off

Also included a tiny CSS fix for "instant" links

### CODE REVIEW

**Comment by afmicka** [comment]
@vgoodric these changes are breaking preview functionality (and our promo nala tests, thats why it was not passing. We have added those tests recently so please do not skip run-nala label). If you go to the url below and try to set to default any of the manifests and preview, it will not go through. It reloads again with all the manifests set to all. 
https://mepparamtarget--milo--adobecom.hlx.live/drafts/nala/features/promotions/promo-default
Screenshot 2024-04-09 at 09 40 20
Screenshot 2024-04-09 at 09 40 12


It works on: https://main--milo--adobecom.hlx.live/drafts/nala/features/promotions/promo-default

Could you please review? 

cc: @nateekar

**Review by chrischrischris — APPROVED** [review]

**Review by afmicka — CHANGES_REQUESTED** [review]
Please review MEP preview functionality on existing pages with multiple manifests like UK acom homepage as well. The changes are breaking it there too.

https://main--homepage--adobecom.hlx.page/uk/homepage/index-loggedout?milolibs=mepparamtarget

**Comment by vgoodric** [comment]
@afmicka I turned off nala because there was something broken on stage as well.  So it wasn't instructive.  But I think everything is working now.  Can you confirm?

File: libs/features/personalization/personalization.js [hunk 1]
```diff
@@ -680,13 +665,37 @@ export async function runPersonalization(experiment, config) {
...
+  });
+  if (selectedVariant) {
+    if (variantNames.includes(selectedVariant)) {
+      manifest.selectedVariantName = selectedVariant;
+      manifest.selectedVariant = manifest.variants[selectedVariant];
+    } else {
+      manifest.selectedVariantName = selectedVariant;
+      manifest.selectedVariant = manifest.variants[selectedVariant];
+    }
+  }
```
Comment (mokimo) [inline libs/features/personalization/personalization.js:688]: I'd avoid the nested if elses
```suggestion
  if (!selectedVariant) return
  if (variantNames.includes(selectedVariant)) {
    manifest.selectedVariantName = selectedVariant;
    manifest.selectedVariant = manifest.variants[selectedVariant];
    return
  } 
  manifest.selectedVariantName = selectedVariant;
  manifest.selectedVariant = manifest.variants[selectedVariant];
```

**Review by mokimo — APPROVED** [review]

**Comment by afmicka** [comment]
> @afmicka I turned off nala because there was something broken on stage as well. So it wasn't instructive. But I think everything is working now. Can you confirm?

Thanks @vgoodric. Confirming it is ok now.

**Comment by Blainegunn** [comment]
@vgoodric no verified label. Please add

**Pull request merged**